mkosi/__init__.py

```diff
@@ -1703,6 +1706,38 @@ def build_uki(
             ]  # fmt: skip
         else:
             arguments += ["--pcr-private-key", context.config.sign_expected_pcr_key]
+    elif ArtifactOutput.pcrs in context.config.split_artifacts:
+        assert context.config.sign_expected_pcr_certificate
+
+        arguments += [
+            "--json=short",
+            "--policy-digest",
+            "--pcr-banks", "sha256",
+        ]  # fmt: skip
+
+        # If we are given an x509 certificate we need to extract the public key to use it
+        if context.config.sign_expected_pcr_certificate.read_text().startswith("-----BEGIN CERTIFICATE-----"):
+            public_key = context.workspace / "pcr-public-key.pem"
+            run(
+                [
+                    "openssl",
+                    "x509",
+                    "-pubkey",
+                    "-noout",
+                    "-in",
+                    context.config.sign_expected_pcr_certificate,
+                ],
+                stdout=public_key.open("wb"),
+            )
+            arguments += ["--pcr-public-key", workdir(public_key)]
+            options += ["--ro-bind", public_key, workdir(public_key)]
```

Comment:
Use `systemd-keyutil public` please, and pass in all the proper arguments so it works with providers as well